src/mpid/ch4/shm/posix/eager/fbox/func_table.c

```diff
@@ -8,18 +8,29 @@
  *  Contributor License Agreement dated February 8, 2012.
  */
 
-#ifndef POSIX_EAGER_INLINE
+/* This requires some explanation: when inline is disabled we need
+ * to avoid duplication of MPIDI_POSIX_eager_send/recv symbols by
+ * posix_eager_impl.h, here included through mpidimpl.h. This goal
+ * is achieved by defining POSIX_EAGER_DISABLE_INLINES. In this case
+ * the definition of MPIDI_POSIX_eager_send/recv is provided by the
+ * posix_eager_inline.h file. When inline is enabled we don't need
+ * the send/recv provided by posix_eager_impl.h either. */
 #define POSIX_EAGER_DISABLE_INLINES
 
 #include <mpidimpl.h>
 #include <posix_eager.h>
 #include "../../posix_types.h"
-#include "posix_eager_inline.h"
+#include "fbox_noinline.h"
 
-MPIDI_POSIX_eager_funcs_t MPIDI_POSIX_eager_fbox_funcs = {
+MPIDI_POSIX_eager_init_funcs_t MPIDI_POSIX_eager_fbox_init_funcs = {
```

Comment:
This isn't going to fix multi-shmmod builds, right? This seems like just a hack to un-inline some init code.

Reply:
Do you mean multi-eager protocols?

Reply:
I have only tested fastbox for now. In principle this should work. The only difference is that besides unlinening the fbox code we also move to function tables for both inline and noinline cases for the init/finalize functions.

Reply:
> Do you mean multi-eager protocols?

Sorry, yes, I meant multi-eager builds. Though now that I try it, I am able to build both fbox and stub at the same time with this PR... Maybe I am misunderstanding the infrastructure. I will have to look at on Monday.

Reply:
Is this additional of another function table actually necessary here? Isn't the namespace and macro defines for inline configurations sufficient to allow this code to move out of inline functions?

Reply:
As @minsii noted there is little benefit in avoiding function tables for `init|finalize`. Thus this PR also defaults to function tables for these two regardless of whether inlinining is enabled or not, this is why we need separate function tables. If you remove the last two commits then you should be left with what you are suggesting.

Reply:
The benefit is that code is a little simpler as we don't have to use some trick to redefine init and finalize symbols when inlining is enabled and use guards in some part of the code.

Reply:
But we already use that trick for all the other functions in the eager modules, no?

Reply:
We have only one eager module: `fbox`, now we are adding `iqueue`. I think it is easy to switch to a simpler approach for init/finalize.

Reply:
For any other perf-critical functions, we should still use the trick.

Reply:
I'm disagreeing that an additional function table is simpler. I get init/finalize is off the critical path, but we do not make this differentiation anywhere else in ch4. This is more complex, IMO.

Reply:
I do not have strong preference on always-function-table v.s. trick for init/finalize. We can switch back to the trick version if you think it is good for consistency.

Reply:
Should I get rid of the last two commits then?

Reply:
Yes, please.

Reply:
New version looks good to me. I'm still disturbed by the differences in this infrastructure vs. the netmods, but I don't think that needs to be solved right this moment.